src/workspace/workspace_mut.rs

```diff
@@ -0,0 +1,547 @@
+use crate::cli::cli_config::PrefixUpdateConfig;
+use crate::diff::LockFileDiff;
+use crate::environment::LockFileUsage;
+use crate::lock_file::{LockFileDerivedData, UpdateContext, UpdateMode};
+use crate::workspace::grouped_environment::GroupedEnvironment;
+use crate::workspace::{MatchSpecs, PypiDeps, SourceSpecs, UpdateDeps, NON_SEMVER_PACKAGES};
+use crate::{load_lock_file, Workspace};
+use indexmap::IndexMap;
+use itertools::Itertools;
+use miette::{IntoDiagnostic, NamedSource};
+use pep440_rs::VersionSpecifiers;
+use pep508_rs::Requirement;
+use pep508_rs::VersionOrUrl::VersionSpecifier;
+use pixi_config::PinningStrategy;
+use pixi_manifest::pypi::PyPiPackageName;
+use pixi_manifest::toml::TomlDocument;
+use pixi_manifest::utils::WithSourceCode;
+use pixi_manifest::{
+    DependencyOverwriteBehavior, FeatureName, FeaturesExt, HasFeaturesIter, LoadManifestsError,
+    ManifestDocument, ManifestKind, PypiDependencyLocation, SpecType, TomlError, WorkspaceManifest,
+    WorkspaceManifestMut,
+};
+use pixi_spec::PixiSpec;
+use rattler_conda_types::{NamelessMatchSpec, PackageName, Platform, Version};
+use rattler_lock::LockFile;
+use std::collections::{HashMap, HashSet};
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+use toml_edit::DocumentMut;
+
+struct OriginalContent {
+    manifest: WorkspaceManifest,
+    source: String,
+}
+
+pub struct WorkspaceMut {
```

Comment:
Could you please add a docstring, explaining why this struct exists on top of `Workspace`?